### DISCUSSION TITLE
Syslog over TCP: two packets for each message

### DISCUSSION THREAD

**Discussion by vostrelt** [syslog-ng/syslog-ng#4543] [Q&A] [answered]
Hello,
I send syslog messages using `syslog` driver with `transport(tcp)` and I observe every sent message results in two packets. First one contains only number - length of syslog message. Second one contains the message itself. I guess it means it uses octet-counted framing. Can it be disabled? Or should I use the `network` driver instead?
Thanks, Tomáš

![image](https://github.com/syslog-ng/syslog-ng/assets/114070266/69a3a5ed-0afe-4e61-ae51-eaf6cf851911)

**Comment by bazsi** [comment, marked as answer]
This is the rfc5424 style syslog format and framing. You can get rid of the
message length (which is part of the transport protocol) by using the
network() driver. The network() driver would use the BSD style message by
default, but you can change that to rfc5424 using flags(syslog-protocol).

So

network("ipaddr" flags(syslog-protocol))

Or simply

network("ipaddr") if you prefer BSD message format.

**Reply by vostrelt** [reply]
Thank you for the answer!
